### PR TITLE
Optimize Bulk Load feature for Milvus 2.1

### PR DESCRIPTION
**Pull request by soothing-rain**
/kind improvement
/hold

Signed-off-by: Yuchen Gao <yuchen.gao@zilliz.com>

### CODE REVIEW

**Comment by soothing-rain** [comment]
rerun ut

**Comment by soothing-rain** [comment]
rerun ut

**Comment by soothing-rain** [comment]
rerun ut

**Comment by soothing-rain** [comment]
rerun ut

File: internal/datacoord/services.go [hunk 1]
```diff
@@ -1155,16 +1161,106 @@ func (s *Server) AddSegment(ctx context.Context, req *datapb.AddSegmentRequest)
...
+	}, nil
+}
+
+func (s *Server) CompleteBulkLoad(ctx context.Context, req *datapb.CompleteBulkLoadRequest) (*commonpb.Status, error) {
+	log.Info("received CompleteBulkLoad request", zap.Int64("task ID", req.GetTaskId()))
+	// Check index status.
+	checkIndexStatus, err := s.rootCoordClient.CheckSegmentIndexReady(ctx, &internalpb.CheckSegmentIndexReadyRequest{
+		TaskID: req.GetTaskId(),
+		ColID:  req.GetCollectionId(),
+		SegIDs: req.GetSegmentIds(),
```
[the remark this answers is not in the document]
Comment (xiaofan-luan) [reply internal/datacoord/services.go:1225]: so the segment state path will be 
growing -> sealed -> flushed -> indexed -> dropped

File: internal/proxy/impl.go [hunk 2]
```diff
@@ -3647,10 +3648,35 @@ func (node *Proxy) Import(ctx context.Context, req *milvuspb.ImportRequest) (*mi
...
+		log.Error("task not persisted yet after wait limit",
+			zap.Int64("wait limit (seconds)", int64(CheckTaskPersistedWaitLimit.Seconds())),
+			zap.Int64("task ID", taskID),
+			zap.Any("current task state", getImportResp.GetState()))
+		return
+	}
+
+	// Start background process of complete bulk load operation.
+	// Ignoring complete bulk load return status.
+	node.dataCoord.CompleteBulkLoad(node.ctx, &datapb.CompleteBulkLoadRequest{
```
Comment (xiaofan-luan) [inline internal/proxy/impl.go:3672]: Doesn't make sense to me.
If impormanager is at Rootcoord, why proxy go to datacoord directly to complete Bulkload?
What if proxy crashed?

**Pull request closed without merging**